### ISSUE TITLE
`MapLibreMap.OnMoveListener::onMoveBegin` is also called on tap

### ISSUE THREAD

**Issue by westnordost** [maplibre/maplibre-native#2792]
**Describe the bug**
`onMoveBegin(detector: MoveGestureDetector)` in a `MapLibreMap.OnMoveListener` is also called when the map is not actually moved but when the user just taps the map.

**To Reproduce**
Register an `MapLibreMap.OnMoveListener` and log something on `onMoveBegin`. Start the app and tap the map. Something will be logged.

**Expected behavior**
`onMoveBegin(detector: MoveGestureDetector)` is only called when the map is moved

**Platform information (please complete the following information):**
 - OS: Android, any API level
 - Version 11.2.0

**Additional context**
In StreetComplete, there is a follow-me mode. I.e. the map automatically centers on the current user's location. This is turned off when the user moves the map manually but it should **not** turn off when the user just taps the map.

Cause of this issue: https://github.com/streetcomplete/StreetComplete/issues/5824

**Comment by westnordost** [comment]
I noticed that the various `*GestureDetector`s (shove, move, rotate, scale, ...) each have a field named `*threshold` which governs the minimum movement necessary for the gesture detector to detect the movement.

By setting a breakpoint during the map initialization, I noticed that those various `*GestureDetector`s actually do have a minimum default threshold set, although I didn't find where it is set.
It was 43.5px for the shove gesture, 18.9px for the scale gesture, 3° for the rotate gesture **but 0px for the move gesture**.

So, wherever these defaults are set, the move gesture should also get some minimum threshold.

**Comment by louwers** [comment]
Here maybe?

https://github.com/maplibre/maplibre-native/blob/bf40cdf5f73403d9855d02a10fdf5eba0a43cbe0/platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/location/LocationCameraController.java#L317

**Comment by westnordost** [comment]
Not sure, maybe at that point, too.

Reading the code vertically, it looks like the `LocationCameraController` is only used at all when the `LocationComponent` is used. But the usage of the location component is optional, as far as I know.

**Comment by westnordost** [comment]
Gesture detector defaults are (first) set in the constructor of `AndroidGesturesManager`, that's where the previously mentioned values 43.5px for the shove gesture, 18.9px for the scale gesture come from:

https://github.com/maplibre/maplibre-gestures-android/blob/main/library/src/main/java/org/maplibre/android/gestures/AndroidGesturesManager.java#L144

In MapLibre-native, `MapGestureDetector` creates a `AndroidGesturesManager` in its constructor and then configures it. It sets the angle threshold for the rotate gesture detector to 3°.

https://github.com/maplibre/maplibre-native/blob/main/platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/maps/MapGestureDetector.java#L173

So, anything else not set, is 0. A threshold for the MoveGestureDetector is not set in any of these places.

Now, the `LocationCameraController`, a component of `LocationComponent` can overwrite the default gestures manager with a modified one. What does it modify? It apparently modifies the move gesture detector move threshold to to something non-`0` (25dp) when the camera ought to follow the user's location and resets to 0 when the camera is not following the user's location anymore.

25dp seems quite large, but I guess this is to make the camera-following a bit "sticky", i.e. it requires a larger move to un-stick the camera-following from the user's location.

---

@louwers 
So, after this research, I suggest the following fix (I can do a PR if you agree):

**Set a certain small move threshold by default, e.g. 5dp** , so to fix this issue by not triggering the move gesture on a single tap but also to not noticeably delay the move gesture (through a large move threshold).

Currently StreetComplete has a default move threshold of 5dp, you can try it out how it feels. You will only notice it if you really really really look for it. I think any non-0 value should probably not trigger the issue that any tap is registered as a move.

Implementation would...

1.

Add 
```xml
<dimen name="maplibre_move_threshold">5dp</dimen>
```
to dimens.xml

2.

Add 

```java
val moveThreshold = context.getResources().getDimension(R.dimen.maplibre_move_threshold);
gesturesManager.getMoveGestureDetector().setMoveThreshold(moveThreshold);
```

to [`MapGestureDetector::initializeGesturesManager`](https://github.com/maplibre/maplibre-native/blob/main/platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/maps/MapGestureDetector.java#L173)

(or alternatively to `AndroidGesturesManager::initDefaultThresholds` if you think this is cleaner)

3.

Replace

```diff
- moveGestureDetector.setMoveThreshold(0f);
+ moveGestureDetector.setMoveThreshold(initialGesturesManager.getMoveGestureDetector().getMoveThreshold())
```

(This is kind of a separate bug, that the LocationCameraController just overwrites the default move threshold whenever it is not actively following the user's location)

**Comment by louwers** [comment]
Makes sense. That is definitely not expected behavior I think.  I think it's fair to categorize this as a bugfix instead of a breaking change. PR welcome!

Could you add a changelog entry with a line explaining the change and mention how people can set it back to the old behavior?
